src/pdsfile/holdings_maintenance/pds3/pdschecksums.py

```
@@ -576,282 +574,48 @@ def update(pdsdir, selection=None, *, logger=None, limits=None):
-            proceed = False
+def main():
+    result = _shelf_common.run_selection_main(SPEC, TASKS, sys.argv)
 
     # If everything went well, execute pdsinfoshelf too
-    if proceed and args.infoshelf:
+    if result.proceed and result.args.infoshelf:
         new_list = [a.replace('pdschecksums', 'pdsinfoshelf') for a in sys.argv]
         new_list = [a for a in new_list if a not in ('--infoshelf', '-i')]
-        status = os.system(' '.join(new_list))
-        sys.exit(status)
+        completed = subprocess.run(new_list, check=False)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)